nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/drive/ListGoogleDrive.java

```diff
@@ -0,0 +1,351 @@
+                || FOLDER_ID.equals(property)
+                || RECURSIVE_SEARCH.equals(property);
+    }
+
+    @Override
+    protected Scope getStateScope(final PropertyContext context) {
+        return Scope.CLUSTER;
+    }
+
+    @Override
+    protected RecordSchema getRecordSchema() {
+        return GoogleDriveFileInfo.getRecordSchema();
+    }
+
+    @Override
+    protected String getDefaultTimePrecision() {
+        return PRECISION_SECONDS.getValue();
+    }
+
+    @Override
+    protected List<GoogleDriveFileInfo> performListing(
+            final ProcessContext context,
+            final Long minTimestamp,
+            final ListingMode listingMode
+    ) throws IOException {
+        final List<GoogleDriveFileInfo> listing = new ArrayList<>();
+
+        final String folderId = context.getProperty(FOLDER_ID).evaluateAttributeExpressions().getValue();
+        final Boolean recursive = context.getProperty(RECURSIVE_SEARCH).asBoolean();
+        final Long minAge = context.getProperty(MIN_AGE).asTimePeriod(TimeUnit.MILLISECONDS);
+
+        Drive driveService = new Drive.Builder(
+                this.httpTransport,
+                JSON_FACTORY,
+                getHttpCredentialsAdapter(
+                        context,
+                        Arrays.asList(DriveScopes.DRIVE_METADATA_READONLY)
+                )
+        )
+                .setApplicationName(APPLICATION_NAME)
+                .build();
+
+        StringBuilder queryBuilder = new StringBuilder();
+        queryBuilder.append(buildQueryForDirs(driveService, folderId, recursive));
+        queryBuilder.append(" and (mimeType != 'application/vnd.google-apps.folder')");
+        queryBuilder.append(" and (mimeType != 'application/vnd.google-apps.shortcut')");
+        queryBuilder.append(" and trashed = false");
+        if (minTimestamp != null && minTimestamp > 0) {
+            String formattedMinTimestamp = DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(OffsetDateTime.ofInstant(Instant.ofEpochMilli(minTimestamp), ZoneId.of("UTC")));
+
+            queryBuilder.append(" and (");
+            queryBuilder.append("modifiedTime >= '" + formattedMinTimestamp + "'");
+            queryBuilder.append(" or createdTime >= '" + formattedMinTimestamp + "'");
+            queryBuilder.append(")");
+        }
+        if (minAge != null && minAge > 0) {
+            long maxTimestamp = System.currentTimeMillis() - minAge;
+            String formattedMaxTimestamp = DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(OffsetDateTime.ofInstant(Instant.ofEpochMilli(maxTimestamp), ZoneId.of("UTC")));
+
+            queryBuilder.append(" and modifiedTime < '" + formattedMaxTimestamp + "'");
+            queryBuilder.append(" and createdTime < '" + formattedMaxTimestamp + "'");
+        }
+
+        String pageToken = null;
+        do {
+            FileList result = driveService.files()
+                    .list()
+                    .setQ(queryBuilder.toString())
```

Comment:
There seems to be a limit for the length of the query string: 30000 characters. It means we can add 500-550 folders in this query.
It should be documented for the user that they cannot list Google Drive shares with more than 500 subfolders.